Sources/Sentry/Public/SentryDebugImageProvider.h

```diff
@@ -7,30 +7,68 @@ NS_ASSUME_NONNULL_BEGIN
 
 /**
  * Reserved for hybrid SDKs that the debug image list for symbolication.
+ * @todo This class should be renamed to @c SentryDebugImage in a future version.
  */
 @interface SentryDebugImageProvider : NSObject
 
 - (instancetype)init;
 
 /**
  * Returns a list of debug images that are being referenced in the given threads.
- * @param threads A list of @c SentryThread that may or may not contain stacktracse.
+ * @param threads A list of @c SentryThread that may or may not contain stacktraces.
+ * @warning This assumes a crash has occurred and attempts to read the crash information from each
```

Comment:
``m`: Should we maybe deprecate this method then?

Reply:
I thought about that. I worried about inserting a deprecation notice in case people use warnings-as-errors, then we'd break their build. I'm ok to do it though, just let me know.

Reply:
I'm pretty sure almost nobody uses this API except for hybrid SDKs controlled by us. It even says so in the comment:https://github.com/getsentry/sentry-cocoa/blob/4053ee90b75473db21ccfc2e1f361f22e2d0544c/Sources/Sentry/Public/SentryDebugImageProvider.h#L8-L11

So I think it's fine to deprecate these methods. Actually, this class shouldn't be public. I created an issue for that https://github.com/getsentry/sentry-cocoa/issues/3056

Reply:
Cool, i'm good with that! Will add.

Reply:
Done in https://github.com/getsentry/sentry-cocoa/pull/3042/commits/58921ff1664f6d40bbfefde567d15fea25545b63. Also added a note to remove them in https://github.com/getsentry/sentry-cocoa/discussions/2946#discussioncomment-5988220.